### PR TITLE
Update to Transfers documentations

### PR DESCRIPTION
**Pull request by asaulenko**

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]
nice work

you could add on the required part a note:

Note: These have been replaced by "orFail" methods (see below).

File: 202009.0/5dc7f1b6-3cef-44e1-8302-5d4e23e0cf0c.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ $cartTransfer->addItem($cartItem1)->addItem($cartItem2);
 $items = $cartTransfer->getItems();
 ```
 
-### Checking the Required Fields
+### Checking the Required Fields (deprecated in version 3.25.0)
```
Comment (gechetspr) [inline 202009.0/5dc7f1b6-3cef-44e1-8302-5d4e23e0cf0c.md:136]: I know that this is quite obvious thing, but this document about transfer feature and not Transfer module. Would be nice to have module name near the required version

File: 202009.0/5dc7f1b6-3cef-44e1-8302-5d4e23e0cf0c.md [hunk 2]
```diff
@@ -142,6 +142,20 @@ In general a transfer object must not know which fields are required, because it
 $customerTransfer->requireFirstName()->getFirstName();
 ```
 
+### Using Get-or-Fail Methods
+Starting from version 3.25.0 transfers expose new get-or-fail methods that are meant to replace the deprecated require-methods. Here is how it works: for each nullable property an extra-getter-method is generated, along with the regular one. This new getter will throw an exception when trying to get the value of a property which was not previously set. The name of the method is composed of `get` prefix, the name of the property and the `OrFail` suffix.
```
Comment (gechetspr) [inline 202009.0/5dc7f1b6-3cef-44e1-8302-5d4e23e0cf0c.md:146]: same

**Review by gechetspr — COMMENTED** [review]
Only 2 comments. The rest looks clear and straightforward to me

**Review by gechetspr — APPROVED** [review]

**Pull request merged**